File: core/src/com/alta_v2/physicsModule/executionContext/Tenant.java
package com.alta_v2.physicsModule.executionContext;

import lombok.RequiredArgsConstructor;

import java.util.UUID;

/**
 * Provides facade of reservable data.
 */
@RequiredArgsConstructor
public class Tenant {

    private final String alias;
    private final String salt = UUID.randomUUID().toString();

    /**
     * Gets the unique identifier of tenant.
     */
    public String getId() {
        return this.alias + "_" + salt;
    }

}

File: core/src/com/alta_v2/physics/executionContext/Tenant.kt
package com.alta_v2.physics.executionContext

import java.util.*

/**
 * Provides facade of reservable data.
 */
class Tenant(private val alias: String) {

    private val salt = UUID.randomUUID().toString()

    /**
     * Gets the unique identifier of tenant.
     */
    val id: String
        get() = alias + "_" + salt
}